numpy/__init__.pyi

```diff
@@ -1231,3 +1231,113 @@ def amin(
     initial: _NumberLike = ...,
     where: _ArrayLikeBool = ...,
 ) -> Union[number, ndarray]: ...
+
+# TODO: `np.prod()``: For object arrays `initial` does not necasarily
+# have to be a numerical scalar.
+# The only requirement is that it is compatible
+# with the `.__mult__()` method(s) of the passed array's elements.
+
+# Note that the same holds for `np.sum()` and `.__add__()`.
```

Comment:
This applies generally to `ufunc.reduce`, for which `np.sum` (`np.add.reduce`) and `np.prod` (`np.multiply.reduce`) are just wrappers.

Reply:
addressed in https://github.com/numpy/numpy/pull/16729/commits/62f26cf464442379219a8e8b04f7e42c8f41340d